File: second-year/second-semester/MPP/mpp-proiect-repository-raulbaciulescu/travelAgency2/model/Purchase.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using model;

namespace model
{
    [Serializable]
    public class Purchase : Entity<long>
    {

        public Flight flight { get; set; }
        public string clientName { get; set; }
        public string clientAddress { get; set; }
        public List<string> tourists { get; set; }
        public int nrOfSeats { get; set; }

        public Purchase(long id, Flight flight, string clientName, string clientAddress, List<string> tourists, int nrOfSeats) : base(id)
        {
            this.flight = flight;
            this.clientName = clientName;
            this.clientAddress = clientAddress;
            this.tourists = tourists;
            this.nrOfSeats = nrOfSeats; 
        }
        
        public Purchase(Flight flight, string clientName, string clientAddress, List<string> tourists, int nrOfSeats) : base(0L)
        {
            this.flight = flight;
            this.clientName = clientName;
            this.clientAddress = clientAddress;
            this.tourists = tourists;
            this.nrOfSeats = nrOfSeats; 
        }
        
        public override string ToString()
        {
            return base.Id + " " + flight + " " + clientName + " " + clientAddress;
        }

        public override bool Equals(object obj)
        {
            return obj is Purchase purchase &&
                   Id == purchase.Id &&
                   EqualityComparer<Flight>.Default.Equals(flight, purchase.flight) &&
                   clientName == purchase.clientName &&
                   clientAddress == purchase.clientAddress &&
                   EqualityComparer<List<string>>.Default.Equals(tourists, purchase.tourists) &&
                   nrOfSeats == purchase.nrOfSeats;
        }

        public override int GetHashCode()
        {
            int hashCode = -1198551226;
            hashCode = hashCode * -1521134295 + Id.GetHashCode();
            hashCode = hashCode * -1521134295 + EqualityComparer<Flight>.Default.GetHashCode(flight);
            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(clientName);
            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(clientAddress);
            hashCode = hashCode * -1521134295 + EqualityComparer<List<string>>.Default.GetHashCode(tourists);
            hashCode = hashCode * -1521134295 + nrOfSeats.GetHashCode();
            return hashCode;
        }
    }
}

File: second-year/second-semester/MPP/mpp-proiect-repository-raulbaciulescu/travelAgency/Model/src/main/java/domain/Purchase.java
package domain;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

public class Purchase extends Entity<Long> implements Serializable {
    private Flight flight;
    private String clientName;
    private String clientAddress;
    private List<String> tourists;
    private int nrOfSeats;

    public Purchase(Flight flight, String clientName, String clientAddress,
                    List<String> tourists, int nrOfSeats) {
        this.flight = flight;
        this.clientName = clientName;
        this.clientAddress = clientAddress;
        this.tourists = tourists;
        this.nrOfSeats = nrOfSeats;
    }

    public Purchase(Flight flight, String clientName, String clientAddress, int nrOfSeats) {
        this.flight = flight;
        this.clientName = clientName;
        this.clientAddress = clientAddress;
        this.nrOfSeats = nrOfSeats;
        tourists = new ArrayList<>();
    }

    public Flight getFlight() {
        return flight;
    }

    public void setFlight(Flight flight) {
        this.flight = flight;
    }

    public String getClientName() {
        return clientName;
    }

    public void setClientName(String clientName) {
        this.clientName = clientName;
    }

    public String getClientAddress() {
        return clientAddress;
    }

    public void setClientAddress(String clientAddress) {
        this.clientAddress = clientAddress;
    }

    public List<String> getTourists() {
        return tourists;
    }

    public void setTourists(List<String> tourists) {
        this.tourists = tourists;
    }

    public int getNrOfSeats() {
        return nrOfSeats;
    }

    public void setNrOfSeats(int nrOfSeats) {
        this.nrOfSeats = nrOfSeats;
    }

    @Override
    public String toString() {
        return "Purchase{" +
                "flight=" + flight +
                ", clientName='" + clientName + '\'' +
                ", clientAddress='" + clientAddress + '\'' +
                ", tourists=" + tourists +
                ", nrOfSeats=" + nrOfSeats +
                '}';
    }
}
